core/unit_test/TestViewCtorDimMatch.hpp

```diff
@@ -47,384 +47,150 @@
+      EXPR,                                                                   \
+      "Constructor for Kokkos::View 'v' has mismatched number of arguments. " \
+      "The number of arguments = " +                                          \
+          std::to_string(ARGS) +                                              \
+          " neither matches the dynamic rank = " + std::to_string(DYNRANK) +  \
+          " nor the total rank = " + std::to_string(TOTALRANK))
+
+template <int rank, int dynrank, template <int> class RankType,
+          std::size_t... args>
+void test_matching_arguments_rank_helper(std::index_sequence<args...>) {
+  constexpr int nargs = sizeof...(args);
+  using view_type     = Kokkos::View<typename RankType<rank>::type>;
+  if (nargs == rank || nargs == dynrank)
+    LIVE({ view_type v("v", args...); }, nargs, dynrank, rank);
+  else
+    DIE({ view_type v("v", args...); }, nargs, dynrank, rank);
+}
 
-#define PARAM_0
-#define PARAM_1 1
-#define PARAM_2 1, 1
-#define PARAM_3 1, 1, 1
-#define PARAM_4 1, 1, 1, 1
-#define PARAM_5 1, 1, 1, 1, 1
-#define PARAM_6 1, 1, 1, 1, 1, 1
-#define PARAM_7 1, 1, 1, 1, 1, 1, 1
+template <int rank, int dynrank, template <int> class RankType>
+void test_matching_arguments_rank() {
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1, 1>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1, 1, 1>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1, 1, 1, 1>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1, 1, 1, 1, 1>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1, 1, 1, 1, 1, 1>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1, 1, 1, 1, 1, 1, 1>());
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<1, 1, 1, 1, 1, 1, 1, 1>());
+}
 
-#define PARAM_0_RANK 0
-#define PARAM_1_RANK 1
-#define PARAM_2_RANK 2
-#define PARAM_3_RANK 3
-#define PARAM_4_RANK 4
-#define PARAM_5_RANK 5
-#define PARAM_6_RANK 6
-#define PARAM_7_RANK 7
+template <int rank>
+struct DynamicRank {
+  using type = typename DynamicRank<rank - 1>::type*;
+};
 
-using DType = int;
+template <>
+struct DynamicRank<0> {
+  using type = int;
+};
 
 // Skip test execution when KOKKOS_ENABLE_OPENMPTARGET is enabled until
 // Kokkos::abort() aborts properly on that backend
 // Skip test execution when KOKKOS_COMPILER_NVHPC until fixed in GTEST
-#if defined(KOKKOS_ENABLE_OPENMPTARGET) || (KOKKOS_COMPILER_NVHPC)
+#if defined(KOKKOS_ENABLE_OPENMPTARGET) || defined(KOKKOS_COMPILER_NVHPC)
```

Comment:
I suspect the OpenMPTarget guards are not needed and that we can compile and just skip execution (if needed at all)